File: COURS/C/TP-06/banking_program.c
#include <stdio.h>

// Function declarations
void checkBalance(float balance);
float deposit();
float withdraw(float balance);

// Banking program in C
int main()
{
    int choice = 0;
    float balance = 0.0f;

    printf("*** Welcome to the Bank ***");

    do
    {
        // Display menu
        printf("\n\n-- Select an option: --\n");
        printf("1. Check balance\n");
        printf("2. Deposit money\n");
        printf("3. Withdraw money\n");
        printf("4. Exit\n");

        printf("Enter your choice: ");
        scanf("%d", &choice);

        // Handle user choice
        switch (choice)
        {
        case 1:
            checkBalance(balance);
            break;
        case 2:
            balance += deposit();
            break;
        case 3:
            balance -= withdraw(balance);
            break;
        case 4:
            printf("\nThank you for using the bank!\n");
            break;
        default:
            printf("\nInvalid choice. Please select 1 - 4.\n");
            break;
        }
    } while (choice != 4);

    return 0;
}

// Function to display the current balance
void checkBalance(float balance)
{
    printf("Your current balance is: $%.2f\n", balance);
}

// Function to handle deposit
float deposit()
{
    float amount = 0.0f;
    printf("Enter the amount to deposit: $");
    scanf("%f", &amount);

    if (amount <= 0)
    {
        printf("Invalid amount! Deposit must be positive.\n");
        return 0.0f;
    }

    printf("Successfully deposited $%.2f\n", amount);
    return amount;
}

// Function to handle withdrawal
float withdraw(float balance)
{
    float amount = 0.0f;
    printf("Enter the amount to withdraw: $");
    scanf("%f", &amount);

    if (amount <= 0)
    {
        printf("Invalid amount. Withdrawal must be positive.\n");
        return 0.0f;
    }
    else if (amount > balance)
    {
        printf("Insufficient balance.\n");
        return 0.0f;
    }

    printf("Successfully withdrawn $%.2f\n", amount);
    return amount;
}

File: COURS/JAVA/TP-05/BankingProgram.java
import java.util.Scanner;

public class BankingProgram {
    // Create a Scanner object to read user input
    static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {

        // Initialize starting balance
        double balance = 10000;

        // Flag to control the loop
        boolean isRunning = true;

        // Variable to store user choice
        int choice;

        // Main menu loop
        while (isRunning) {
            // Display menu
            System.out.println("***** WELCOME TO YOUR BANK ACCOUNT *****");
            System.out.println("1. Check balance");
            System.out.println("2. Deposit money");
            System.out.println("3. Withdraw money");
            System.out.println("4. Exit");

            // Prompt for user input
            System.out.print("Choose an option (1 - 4): ");
            choice = scanner.nextInt();

            // Perform action based on user choice
            switch (choice) {
                case 1 -> showBalance(balance);              // Show balance
                case 2 -> balance += deposit();              // Add deposit to balance
                case 3 -> balance -= withdraw(balance);      // Subtract withdrawal from balance
                case 4 -> isRunning = false;                 // Exit the loop
                default -> System.out.println("Invalid choice."); // Handle invalid input
            }
        }

        // Close the scanner after use
        scanner.close();
    }

    // Method to show current balance
    static void showBalance(double balance) {
        System.out.println("Your balance is: " + balance);
    }

    // Method to handle deposit
    static double deposit() {
        double amount;

        System.out.print("Enter the amount to be deposited: ");
        amount = scanner.nextDouble();

        // Validate deposit amount
        if (amount < 0) {
            System.out.println("Amount can't be negative.");
            return 0;
        } else {
            System.out.println(amount + "$ added.");
            return amount;
        }
    }

    // Method to handle withdrawal
    static double withdraw(double balance) {
        double amount;

        System.out.print("Enter the amount to be withdrawn: ");
        amount = scanner.nextDouble();

        // Validate withdrawal amount
        if (amount > balance) {
            System.out.println("INSUFFICIENT FUNDS");
            return 0;
        } else if (amount < 0) {
            System.out.println("Amount can't be negative.");
            return 0;
        } else {
            System.out.println(amount + "$ withdrawn.");
            return amount;
        }
    }
}
